cmake/macros/macro_deal_ii_setup_target.cmake

```
@@ -53,13 +53,32 @@ macro(deal_ii_setup_target _target)
   endif()
 
   #
-  # Set build type with the help of the specified keyword, or
-  # CMAKE_BUILD_TYPE:
+  # Parse the optional arguments passed to this macro. These
+  # may only contain the build type (DEBUG or RELEASE).
   #
+  set(_argn_build_type)
```

I would simplify the logic by populating the `_build` variable directly:
```suggestion
  set(_build)
```